lib/Doctrine/Common/Inflector/Inflector.php

```diff
@@ -138,7 +138,7 @@ public static function reset() : void
      * }}}
      *
      * @param string  $type         The type of inflection, either 'plural' or 'singular'
-     * @param array|iterable $rules An array of rules to be added.
+     * @param array|iterable<mixed> $rules An array of rules to be added.
```

Comment:
Are the rules really of type mixed?

Reply:
Well it's a deprecated function and is not going to be used anyways.
This change just fixes phpstan warns with `--level max`.

Reply:
Better to hide the error than to document with the incorrect type IMO.

Reply:
Ok.
I went deeper through the code base history and based on some old tests, the exact type is `array<string,string>|iterable<array<string,string>>`.
It is a mapping from _preg patterns_ to _replacements_.
IMO it can be written as `mixed` since it does not really matter because of deprecation.